src/StreamJsonRpc.Shared/DataContracts/JsonRpcMessage.cs

```diff
@@ -67,7 +67,14 @@ public static JsonRpcMessage CreateRequest(int? id, string @method, IReadOnlyLis
 
         public static JsonRpcMessage CreateRequestWithNamedParameters(int? id, string @method, object namedParameters, JsonSerializer parameterSerializer)
         {
-            return new JsonRpcMessage(method, JToken.FromObject(namedParameters, parameterSerializer), id);
+            if (namedParameters == null)
```

Comment:
By passing in null, the serializer omits the parameter altogether, which is allowed according to the spec.  From the common language server protocol, it looks like there can be requests where the parameter is "void".  I tested not passing in any params (passing null in streamjsonprc api), or passing an empty array, against the Java language server.  It seemed to have accepted both.  However, wouldn't we want to support calling a method with no "params" at all?